src/components/Proposal/ProposalContainer.tsx

```diff
@@ -271,6 +251,28 @@ class ProposalContainer extends React.Component<IProps, IState> {
           {isRedeemPending ? <strong><i>Warning: Redeeming for this proposal is already in progress</i></strong> : ''}
         </div>;
 
+      const redeemButton = (redeemable || executable ?
```

Comment:
should be `currentRedemptions || beneficiaryRedemptions || executable` because there may be nonredeemable rewards and we still want to show the button.

Reply:
i think that was what was causing the issue though with showing an execute button when we shouldnt have. ok ill test some more

Reply:
@dev-matan-tsuberi i think it should be `currentRedemptions || beneficiaryHasRewards || executable` because we don't want to show the button if the beneficiary has GP rewards, only CR rewards (beneficiary rewards)

Reply:
:+1: